hack/update-codegen.sh

```diff
@@ -11,16 +11,21 @@ export GOFLAGS=-mod=
 REPO_ROOT=$(dirname "${BASH_SOURCE[@]}")/..
 CODEGEN_PKG=${CODEGEN_PKG:-"${REPO_ROOT}/vendor/k8s.io/code-generator"}
 
+# shellcheck disable=SC1091
+source "${CODEGEN_PKG}/kube_codegen.sh"
+
 # shellcheck disable=SC1091
 source "${REPO_ROOT}"/vendor/knative.dev/hack/codegen-library.sh
 
 KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-"${REPO_ROOT}/vendor/knative.dev/pkg"}
 
 # Generate our own client for Openshift (otherwise injection won't work)
-"${CODEGEN_PKG}/generate-groups.sh" "client,informer,lister" \
-  github.com/openshift-knative/serverless-operator/pkg/client github.com/openshift/api \
-  "route:v1 config:v1" \
-  --go-header-file "${REPO_ROOT}/hack/boilerplate/boilerplate.go.txt"
+kube::codegen::gen_client \
+  --boilerplate "${REPO_ROOT_DIR}/hack/boilerplate/boilerplate.go.txt" \
+  --output-dir "${REPO_ROOT_DIR}/pkg/client" \
+  --output-pkg "github.com/openshift/api" \
+  --with-watch \
+  "${REPO_ROOT_DIR}/pkg/apis"
```

Comment:
I think we can use directly their provided generated code:
https://github.com/openshift/client-go/tree/master/route/informers/externalversions
https://github.com/openshift/client-go/tree/master/config/informers/externalversions

We optimized that upstream in some cases e.g. cert-manager. See https://github.com/knative/serving/pull/15703.

`"${REPO_ROOT_DIR}/pkg/apis"` this causes 
`grep: .../github.com/openshift-knative/serverless-operator/pkg/apis: No such file or directory`, I guess it is a dummy thing to avoid the failure.

Reply:
Great point, strangely I can't reproduce the issue locally with codegen script. 

Let my try to optimize the openshift/client-go bits.

Reply:
It's quiet hefty change, but yes we can get rid of our generated client. With k8s codegen changes by default sourcing from `vendor/` it'd generate client code for other resource we use. 

The other option is implemented below. It's a bit tedious on naming side, but we have injection for specific resource groups now.

https://github.com/openshift-knative/serverless-operator/pull/3375/commits/3c3362680006989064487016b369607ef393bcbc